### ISSUE TITLE
Blinded PSET incompatibility with elementsd

### ISSUE THREAD

**Issue by pavel-kokolemin** [ElementsProject/rust-elements#101]
When I create and blind PSET with two or more outputs the result can't be loaded in `elementsd`:

```
elements-cli analyzepsbt $(cat pset.txt)
error code: -22
error message:
TX decode failed Size of value was not the stated size: iostream error
```

I'm attaching small rust program to reproduce the issue and resulted PSET:
[pset.txt](https://github.com/ElementsProject/rust-elements/files/7521272/pset.txt)
[pset_test.zip](https://github.com/ElementsProject/rust-elements/files/7521273/pset_test.zip)
Same program generates valid PSET when there is only one output.

I tested with latest release (`0.21.0`) and latest dev version of `elements` (commit `04cedf3f6be9e8eeb76af6206656ad2516d16f5a`) and 
latest `rust-elements` (commit `53d5729f8cdc3623f9f110edba86aeaa9059af5b`).
I don't know if the issue on `elements` or ` rust-elements` side.

Thank you for the very good library!

**Comment by sanket1729** [comment]
Hello, first off. Thanks for the minimal stand alone example. This really helped me find out what the problem is. There are a couple of issues here.

First bug: there is a bug in `elementsd` (c++ codebase) global scalar decoding that is causing the issue here. 

Second bug: In rust-elements, the scalars must be cleared after the blinding is done. This is not being done currently and is updated in #100.

I just tested your code by replacing dependencies with 
```
elements = { package = "elements", git = "https://github.com/sanket1729/rust-elements", branch = "update_pset" }
```
and I am able to analyzepsbt with it.

This means that currently:
- Semi-blinded psets (ones after blind(), but not blind_last()) are correctly implemented in rust-elements, but not in elementsd
- Fully blinded psets(ones after calling blind_last()) are incorrectly implemented rust-elements, but will be fixed as of #100. There are a few other updates to the spec, so I suggest prototyping with #100 till we have another release.

So, if you are completely blinding using rust-elements, I think #100 should solve all your issues.

**Comment by pavel-kokolemin** [comment]
Works nicely now. Thank you, Sanket!
